### PR TITLE
Refactor/user profile

### PR DESCRIPTION
**Pull request by frontr-uk**
- Refactor Auth state
- Create state for auth
- Create getUser from API -> 3rd party service

### CODE REVIEW

File: api/services/rdProfessionals.ts [hunk 1]
```diff
@@ -4,7 +4,7 @@ import {http} from '../lib/http';
 export async function getOrganisationId(details) {
   // TODO remove the hardcoded email when correct user gets returned from idam
   // const email = details.data.email;
-  const email = 'henry_fr_harper@yahoo.com'
-  return await http.get(`${config.services.rdProfessionalApi}/search/organisations/${email}`)
+  const email = 'juitestuser2@gmail.com'
+  return await http.get(`${config.services.rd_professional_api}/search/organisations/${email}`)
```
Comment (fazanki) [inline api/services/rdProfessionals.ts]: please revert as per alan we are using camelCase rather the under_score

File: api/user/index.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  //   const response = await http.get(`${config.services.rd_professional_api}/organisations/${req.session.auth.orgId}/users/${req.session.auth.userId}`)
+  //   logger.info('Calling responser', response)
+  //   res.send(response)
+  // } catch (error) {
+  //   logger.info(error)
+  //   //const errReport = JSON.stringify({ apiError: error, apiStatusCode: error.statusCode, message: '3rd party service payment api return error'})
+  //   res.status(500).send(error)
+  // }
+}
+
```
Comment (fazanki) [inline api/user/index.ts]: do u need to remove

File: api/user/user.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  jti: string
+  loa: number
+  sub: string
+  surname: string
+  type: string
+  constructor(prop) {
+    Object.assign(this, prop)
+  }
+}
+export class User {
```
Comment (fazanki) [inline api/user/user.ts]: what is the difference between usermodel and user? can we name them to have more sense

File: src/auth/store/reducers/index.ts [hunk 4]
```diff
@@ -1 +1,66 @@
...
+      };
+    }
+
+    case AuthActionTypes.USER_SUCCESS: {
+      const user = new UserModel(action.payload);
+      return {
+        ...state,
+        user,
+        userDetailsLoaded: true,
+        userDetailsLoading: false,
```
Comment (fazanki) [inline src/auth/store/reducers/index.ts]: think we could keep this as loaded loading just for a sake of consistency... don't think we need to be explicit

File: api/services/rdProfessionals.ts (hunk 1 above)
Comment (frontr-uk) [reply api/services/rdProfessionals.ts]: uu missed this one

File: api/user/user.ts (hunk 3 above)
Comment (frontr-uk) [reply api/user/user.ts]: I was just refactoring and didn't look at the naming convention, but 'name' + Model is a good approach

File: src/auth/store/reducers/index.ts (hunk 4 above)
Comment (frontr-uk) [reply src/auth/store/reducers/index.ts]: yeah good point

File: api/auth/index.ts [hunk 5]
```diff
@@ -92,19 +95,23 @@ export async function oauth(req: EnhancedRequest, res: express.Response, next: e
             const accessToken = response.data.access_token
             const details: any = await getUserDetails(accessToken)
 
+          // logger.info('details are', details.data)
+
```
Comment (fazanki) [inline api/auth/index.ts]: remoe

File: api/user/index.ts [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+  const mockUser: UserProfileModel = UserMock
+
+  try {
+    const response = await http.get(`${config.services.rdProfessionalApi}/search/users/henry_fr_harper@yahoo.com`)
+    // const response = await http.get(`${config.services.rdProfessionalApi}/search/users/${req.session.auth.email}`)
+    logger.info('Calling responser', response)
+    res.send(mockUser)
+  } catch (error) {
+    logger.info(error)
+    //const errReport = JSON.stringify({ apiError: error, apiStatusCode: error.statusCode, message: '3rd party service payment api return error'})
```
Comment (fazanki) [inline api/user/index.ts]: see if  can be removed

**Pull request merged**